gnpy/core/utils.py

```diff
@@ -199,3 +199,42 @@ def rrc(ffs, baud_rate, alpha):
     p_inds = np.where(np.logical_and(np.abs(ffs) > 0, np.abs(ffs) < l_lim))
     hf[p_inds] = 1
     return sqrt(hf)
+
+def merge_dicts_recursively(dict1, dict2):
+    """Merge contents of dicts recursively
+
+    >>> d1 = {'params': {'restrictions': {'preamp_variety_list': [], 'booster_variety_list': []}}}
+    >>> d2 = {'params': {'target_pch_out_db': -20}}
+    >>> merge_dicts_recursively(d1, d2)
+    {'params': {'restrictions': {'preamp_variety_list': [], 'booster_variety_list': []}, 'target_pch_out_db': -20}}
+
+    >>> d3 = {'params': {'restrictions': {'preamp_variety_list': ['foo'], 'booster_variety_list': ['bar']}}}
+    >>> merge_dicts_recursively(d1, d3)
+    {'params': {'restrictions': {'preamp_variety_list': [], 'booster_variety_list': []}}}
```

Comment:
@EstherLerouzic , can you please confirm that this is a correct behavior? I don't know whether it would make more sense to merge the leaf lists or not. Either way, this test just documents the current behavior.

Reply:
Thanks for the commits !
This is the correct behaviour.  Restriction read from node overrides restriction from eqpt_config.json if they exist, so we should not merge preamp and booster lists.
so maybe the naming is not correct, it is more an update

Reply:
- Please feel free to squash my changes to the commits which introduce that code. Or not, I do not really have a preference here.
- Also feel free to rename this function if you feel like doing that. Perhaps something like `merge_amplifier_restrictions`?